sdk/core/Azure.Core.Experimental/src/DynamicJson.cs

```diff
@@ -96,6 +96,37 @@ private IEnumerable GetEnumerable()
 
         private object? SetProperty(string name, object value)
         {
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+            if (_options.PropertyNameCasing == DynamicJsonNameMapping.None)
```

Comment:
```suggestion
            if (_options.PropertyNameCasing == DynamicJsonNameMapping.None || name.Length  == 1)
```

Reply:
I think I disagree with this - would you not lower-case an uppercase single character if your directive was to make it camelCase?  Do we know if this is defined?  It does seem like a degenerate case...

Reply:
Merging for now, but happy to make changes later if you feel strongly about this.